### ISSUE TITLE
[HOLD for payment 2024-05-02] [Mentions v2] Update auto-complete widget to perform online search when needed

### ISSUE THREAD

**Issue by rlinoz** [Expensify/App#39534]
~HOLD https://github.com/Expensify/App/issues/39532~
~HOLD https://github.com/Expensify/App/issues/39505~

When mentioning a room and the auto-complete widget has less than 5 results we will perform a request to search for reports that we might not have locally, this request will call the command `SearchForRoomsToMention` which will take a "query" and a `policyID` as parameters. The result of the call will be merge into onyx with the other reports and will automatically update the auto-complete widget with the new results.

While performing the search the auto-complete widget must show a loading indicator, as its shown [here](https://docs.google.com/document/d/1g_1xLGO29EV6hbOtiYqX53oof_gr8jbnbkSRPjuA4RI/edit#bookmark=id.u5nhfe1xaxvv)

Details can be found [here](https://docs.google.com/document/d/1g_1xLGO29EV6hbOtiYqX53oof_gr8jbnbkSRPjuA4RI/edit#bookmark=id.7cvphfbqhluh) in the Mentions V2 design doc.

**Comment by war-in** [comment]
Hi, I'm Marcin from SoftwareMansion and I would like to work on this one :) could you please assign me? 🙏

**Comment by rlinoz** [comment]
@war-in keep in mind that we still need this PR https://github.com/Expensify/Auth/pull/10483 to get merged to unblock this.

**Comment by war-in** [comment]
@rlinoz Yes I know :) I'm just starting to investigate it and would like to be assigned to avoid duplication

**Comment by robertKozik** [comment]
> @war-in keep in mind that we still need this PR https://github.com/Expensify/Auth/pull/10483 to get merged to unblock this.

Wondering what's the ETA for this?

**Comment by rlinoz** [comment]
It is in review, I think it will get merged today, but deployed only on monday

**Comment by rlinoz** [comment]
https://github.com/Expensify/Auth/pull/10483 is merged, waiting for deploy now.

**Comment by jliexpensify** [comment]
No payment needed right?

**Comment by rlinoz** [comment]
@jliexpensify I think we will have a C+ reviewing the PR later, so  we will need payment at some point

**Comment by war-in** [comment]
@rlinoz @puneetlath We just noticed that there is a bug when trying to click on a single mention suggestion on iOS. After some investigation, it turned out that the issue is connected to `FlashList` usage in the `SuggestionMention`. It is somehow affected by the `FlatList` which is used to display messages. The workaround would be to replace `FlashList` with `FlatList` (it works correctly, but haven't tested it too much).

Bug exists on main branch too so it is not connected to changes from my branch

Example:

https://github.com/Expensify/App/assets/61014013/718131e7-367f-4400-a187-008cc235100a

**Comment by rlinoz** [comment]
Last backend PR is deployed to prod.

**Comment by rlinoz** [comment]
@war-in is that happening only on iOS?

**Comment by rlinoz** [comment]
Anyway, it looks like [there is a lot going on](https://github.com/Expensify/App/issues/40048) with `FlashList` right now, I think it would be better to fix this bug in another issue if it still exists after all the PR's in that issue are merged.

cc: @puneetlath @war-in

**Comment by war-in** [comment]
> is that happening only on iOS?

Yeah, checked on other devices and they are fine.

**Comment by puneetlath** [comment]
Sorry for being dense, I couldn't tell from the video what the bug was. Mind explaining?

**Comment by puneetlath** [comment]
Ohhh I see. Hm, yeah, that's not ideal. But I also agree about handling it separately if that's an existing bug.

**Comment by rlinoz** [comment]
Created an issue here to handle it https://github.com/Expensify/App/issues/40508

**Comment by shawnborton** [comment]
Let me know if anything is needed from Design here.

**Comment by rlinoz** [comment]
Oh I tagged you on the PR https://github.com/Expensify/App/pull/40232#issuecomment-2069147809 sorry

**Comment by shawnborton** [comment]
Sounds good, see ya over there!

**Comment by jliexpensify** [comment]
Ok cool, so this should be the Payment Summary right?

- C+: @parasharrajat $250 (New.Dot request)

@parasharrajat - anything needed for the checklist?

**Comment by war-in** [comment]
@jliexpensify should I create a `regression test`? There is a [comment](https://github.com/Expensify/App/issues/39534#issuecomment-2077213940) from malvin-bot but I haven't seen this before so I need to ask 😅

**Comment by jliexpensify** [comment]
Hi @war-in - sure, if you think one is needed, feel free to propose the steps and I can send this over to QA.

**Comment by puneetlath** [comment]
@rlinoz we should actually probably put together a set of regression tests for room mentions in general.

**Comment by rlinoz** [comment]
Oh interesting, I agree. I will create some steps and share them in slack to get comments.

**Comment by jliexpensify** [comment]
[Payment Summary for Jason Mills (for Rajat)](https://github.com/Expensify/App/issues/39534#issuecomment-2078329566)

**Comment by rlinoz** [comment]
Started a slack thread [here](https://expensify.slack.com/archives/C066HJM2CAZ/p1714397862688059) for regression tests of the entire feature, I will try to open the GH to add the tests by tomorrow.

We can close this one.

**Comment by jliexpensify** [comment]
**NEW PAYMENT SUMMARY**

- C+: @parasharrajat $500 (New.Dot payment)

**Comment by JmillsExpensify** [comment]
$500 approved for @parasharrajat
